File: judge_javascript/lambda.js
const Judge = require("judge");

let successPayload = {
  statusCode: 200,
};

let errorPayload = {
  statusCode: 409,
};

exports.handler = async (event) => {
  try {
    const fileName = event.fileName;
    const problemId = event.problemId;
    const submit = event.submit;

    if (submit === true) {
      await Judge.downloadFile(fileName);
    } else {
      const code = event.code;
      await Judge.createFile(fileName, code);
    }
    const response = await Judge.spawnDFile(fileName, problemId);
    successPayload["body"] = response;
    return successPayload;
  } catch (error) {
    console.log("error: ", error);
    errorPayload["body"] = error.name;
    return errorPayload;
  }
};

File: judge_python/lambda.py
import json
import judge

success_payload = {
    'statusCode': 200,
}

error_payload = {
    'statusCode': 409,
}


def lambda_handler(event, context):
    try:
        file_name = event['fileName']
        problem_id = event['problemId']

        if event['submit']:
            download_file_name = judge.download(file_name)

            if download_file_name != False:
                payload = judge.run_file(download_file_name, problem_id)
                if payload:
                    success_payload['body'] = json.dumps(payload)
                    return success_payload
                else:
                    error_payload['body'] = json.dumps('Error from lambda')
                    return error_payload
        else:
            code = event['code']
            create_file_name = judge.create_file(file_name, code)
            if create_file_name != False:
                payload = judge.run_file(create_file_name, problem_id)
                if payload:
                    success_payload['body'] = json.dumps(payload)
                    return success_payload
                else:
                    error_payload['body'] = json.dumps('Error from lambda')
                    return error_payload
    except Exception as e:
        error_payload['body'] = json.dumps(e)
        return error_payload
